src/adaXT/decision_tree/splitter.pyx

```diff
@@ -115,23 +113,21 @@ cdef class Splitter:
                         current_feature_values[sorted_index_list_feature[i + 1]]):
                     continue
                 # test the split
-                crit, left_imp, right_imp, threshold = self.criteria.evaluate_split(
+                crit, threshold = self.criteria.evaluate_split(
                                                         sorted_index_list_feature, i+1,
                                                         feature
                                                         )
-                if best_score - crit > EPSILON:  # rounding error
+                if best_score > crit:  # rounding error
```

Comment:
with the previous iteration this would not allow for negative values of the best_score. You would have to take abs of the value. It seems to work fine without it, so I have just changed it.

Reply:
Not sure I follow why negative values would not work. Isnt it just the same as best_score > crit +  EPSILON so the EPSILON just ensure that one only updates it if the improvement is better than an rounding error

In general I guess its not super necessary as worst case it just updates the split even though it did not really need to.